cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1320,13 +1419,14 @@ serialiseShelleyBasedTxBody _era txbody txscripts
 
 serialiseShelleyBasedTxBody _era txbody txscripts
                             (TxBodyScriptData _ datums redeemers)
-                            txmetadata =
+                            txmetadata txBodycriptValidity =
     CBOR.serializeEncoding' $
-        CBOR.encodeListLen 5
+        CBOR.encodeListLen 6
      <> CBOR.toCBOR txbody
      <> CBOR.toCBOR txscripts
      <> CBOR.toCBOR datums
      <> CBOR.toCBOR redeemers
+     <> CBOR.toCBOR (txScriptValidityToScriptValidity txBodycriptValidity)
```

Comment:
What will happen here is in a non-Alonzo era we will default to `ScriptValid` which is incorrect.